### DISCUSSION TITLE
How to select rows that can be expanded using getRowCanExpand prop even if renderDetailPanal prop exists

### DISCUSSION THREAD

**Discussion by tacigar** [KevinVandy/material-react-table#144] [Q&A] [answered]
In my use case, even when `renderDetailPanel` prop is specified, some rows need to be unable to be expanded.
I tried to do this using `getRowCanExpand` prop but it did not work.

Looking at the implementation, the reason for this seems to be that the expand button is always enabled if `renderDetailPanel` exists.
https://github.com/KevinVandy/material-react-table/blob/ff91ee55dc1c0ee92e75bd9b4c06fd50cc01b3b5/src/buttons/MRT_ExpandButton.tsx#L44

I would like to work around this behavior somehow.
Do you have any ideas?

**Comment by KevinVandy** [comment, marked as answer]
You can override any expand button props with the `muiExpandButtonProps` prop. You can access the `row` values too within that prop callback.

```tsx
<MaterialReactTable
  columns={columns}
  data={data}
  muiExpandButtonProps={({row}) => ({
    disabled: row.original.whatever //disable when criteria met
  })}
/>
```

**Reply by tacigar** [reply]
Wow! It works!
Thank you so much!
